config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-main-upgrades.yaml

```diff
@@ -1,56 +1,5 @@
 periodics:
 
```

Comment:
I think the following is missing:

For the main branch:
* periodics & presubmits:
  * Bump KUBEBUILDER_ENVTEST_KUBERNETES_VERSION of the test-mink8s jobs to the new minimum supported management cluster version.
* periodics:
  * Bump KUBERNETES_VERSION_MANAGEMENT of the e2e-mink8s job to the new minimum supported management cluster version.


Explanation:

With this PR we are supporting a new Kubernetes version v1.27. This moves our entire support range **on the main branch** up by one.
* mgmt cluster: v1.24-v1.27
* workload cluster: v1.22-v1.27

So KUBEBUILDER_ENVTEST_KUBERNETES_VERSION should be set to: v1.24.2
```
./hack/tools/bin/setup-envtest  list | grep v1.24
(available)  v1.24.2  linux/amd64
(available)  v1.24.1  linux/amd64
```
(we probably should improve the comment above the KUBEBUILDER_ENVTEST_KUBERNETES_VERSION env var here)

KUBERNETES_VERSION_MANAGEMENT should be set to "stable-1.24"

Reply:
Looks perfect, thank you!

> I will squash all commits once reviewers are happy with the state of the patch.

Sounds reasonable :)